snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/tracker/Emitter.java

```diff
@@ -298,15 +325,19 @@ private void buildEmitterUri() {
      *                to be added.
      */
     public void add(final Payload payload) {
-        Executor.execute(new Runnable() {
-            @Override
-            public void run() {
-                eventStore.add(payload);
-                if (isRunning.compareAndSet(false, true)) {
-                    attemptEmit();
+        if (eventStore != null) {
```

Comment:
is there a risk for a race condition? i.e. starting to track events before the events store set up is complete?